src/glove.c

```diff
@@ -348,9 +351,28 @@ int main(int argc, char **argv) {
     int i;
     FILE *fid;
     vocab_file = malloc(sizeof(char) * MAX_STRING_LENGTH);
+    if (NULL == vocab_file){
+        return 1;
+    }
     input_file = malloc(sizeof(char) * MAX_STRING_LENGTH);
+    if (NULL == input_file){
+        free(vocab_file);
+        return 1;
+    }
     save_W_file = malloc(sizeof(char) * MAX_STRING_LENGTH);
+    if (NULL == save_W_file){
+        free(input_file);
+        free(vocab_file);
+        return 1;
+    }
     save_gradsq_file = malloc(sizeof(char) * MAX_STRING_LENGTH);
+    if (NULL == save_gradsq_file){
```

Comment:
can you think of any reason why we should just have

char save_gradsq_file[MAX_STRING_LENGTH]

at the top of the file?

also looking at this it's a little jarring to see sizeof(char).  isn't that always supposed to be 1?  it's been years since i've done any actual C

Reply:
I just committed that fix the missing free. it is always better to use sizeof instead of using constants since the result of the sizeof function depends on the implementation of the compiler and architecture and for clarity of code I would prefer to keep sizeof.

If my commits are helpful, I will continue to investigate it. In the coming days I would like to continue providing support and fixes.